include/crocoddyl/core/integrator/rk4.hxx

```diff
@@ -238,15 +273,41 @@ void IntegratedActionModelRK4Tpl<Scalar>::quasiStatic(const boost::shared_ptr<Ac
 
   // Static casting the data
   const boost::shared_ptr<Data>& d = boost::static_pointer_cast<Data>(data);
-  d->control->w *= 0.;
-  differential_->quasiStatic(d->differential[0], d->control->w, x, maxiter, tol);
-  control_->params(d->control, 0.0, d->control->w);
-  u = d->control->u;
+  d->control[0]->w *= 0.;
+  differential_->quasiStatic(d->differential[0], d->control[0]->w, x, maxiter, tol);
+  control_->params(d->control[0], 0., d->control[0]->w);
+  u = d->control[0]->u;
 }
 
 template <typename Scalar>
 void IntegratedActionModelRK4Tpl<Scalar>::print(std::ostream& os) const {
   os << "IntegratedActionModelRK4 {dt=" << time_step_ << ", " << *differential_ << "}";
 }
 
+template <typename Scalar>
+void IntegratedActionModelRK4Tpl<Scalar>::set_nthreads(const int nthreads) {
```

Comment:
(I may have missed this) - should this be set from/in the problem? Such that we only have to remember to set the threads once?

Reply:
This defines the number of threads used for running internal computations of RK4 in parallel.
This allows us to set different number of threads to the ones used in `ShootingProblem`.

Note that the default number of threads used here (and also in the problem class) is defined by the `CROCODDYL_WITH_NTHREADS` (which is setup in the cmake configuration).